x-pack/plugins/fleet/public/applications/fleet/components/fleet_server_instructions/components/fleet_server_hosts_form.tsx

```diff
@@ -27,14 +29,24 @@ export const FleetServerHostSelect: React.FunctionComponent<FleetServerHostSelec
 }) => {
   const theme = useTheme() as EuiTheme;
 
+  const { cloud } = useStartServices();
+  const isServerless = cloud?.isServerlessEnabled ?? false;
+
   const fleetServerHostsOptions = useMemo(
     () => [
-      ...fleetServerHosts.map((fleetServerHost) => {
-        return {
-          inputDisplay: `${fleetServerHost.name} (${fleetServerHost.host_urls[0]})`,
-          value: fleetServerHost.id,
-        };
-      }),
+      ...fleetServerHosts
+        .filter((fleetServerHost) => {
+          if (!isServerless) {
+            return true;
+          }
+          return fleetServerHost.id === DEFAULT_FLEET_SERVER_HOST_ID; // TODO: is this correct? locally default fs host has id "docker"
```

Comment:
@elastic/fleet Can I get your input on this? I'm not sure if I'm testing this correctly: this constant resolves to `fleet-default-fleet-server-host` but I don't have this id on my local serverless setup. Is this a dev setup issue or is there something else I'm missing? Same question applies to the API handlers.